netty-server/src/main/scala/org/http4s/netty/NettyBuilder.scala

```diff
@@ -254,64 +272,73 @@ class MikuBuilder[F[_]](
   /** A stream transformation that registers our channels and
     * adds the necessary codecs and transport handlers
     */
-  def channelPipe(
+  private[netty] def registerChannel(
       eventLoop: MultithreadEventLoopGroup,
-      allChannels: DefaultChannelGroup): Pipe[F, Channel, Unit] = { s =>
-    s.evalMap { connChannel =>
-      F.delay {
-
-        // Setup the channel for explicit reads
-        connChannel
-          .config()
-          .setOption(ChannelOption.AUTO_READ, java.lang.Boolean.FALSE)
-
-        val pipeline = connChannel.pipeline()
-        getContext() match {
-          case Some((ctx, auth)) => //Ignore client authentication for now
-            val engine = ctx.createSSLEngine()
-            engine.setUseClientMode(false)
-            auth match {
-              case NoClientAuth => ()
-              case ClientAuthRequired =>
-                engine.setNeedClientAuth(true)
-              case ClientAuthOptional =>
-                engine.setWantClientAuth(true)
-            }
-            pipeline.addLast("ssl", new SslHandler(engine))
-
-          case None => //Do nothing
-        }
+      allChannels: DefaultChannelGroup): Channel => F[Unit] = { connChannel =>
+    F.delay {
+
+      // Setup the channel for explicit reads
+      connChannel
+        .config()
+        .setOption(ChannelOption.AUTO_READ, java.lang.Boolean.FALSE)
+
+      val pipeline = connChannel.pipeline()
+      getContext() match {
+        case Some((ctx, auth)) =>
+          val engine = ctx.createSSLEngine()
```

Comment:
I still think passing around an `F[SSLEngine]` might be the right API for our backends.

Reply:
possibly. I've been considering this too and I Want to expose that on master, but for 0.18.x maybe not.